### PR TITLE
Enable Offer To Cluster

### PR DESCRIPTION
**Pull request by Aaronong**
Improve documentation of cluster interfaces, and allow `ClusteredServiceAgent` to offer messages to the cluster itself. 

Comments are copied from Java implementation.

Offer capability modeled after Java Implementation:
- https://github.com/real-logic/aeron/blob/master/aeron-cluster/src/main/java/io/aeron/cluster/service/Cluster.java#L304
- https://github.com/real-logic/aeron/blob/master/aeron-cluster/src/main/java/io/aeron/cluster/service/ClusteredServiceAgent.java#L329
- https://github.com/real-logic/aeron/blob/master/aeron-cluster/src/main/java/io/aeron/cluster/service/ConsensusModuleProxy.java#L117

### CODE REVIEW

File: cluster/cluster.go [hunk 1]
```diff
@@ -1,15 +1,149 @@
...
+	"github.com/lirm/aeron-go/aeron/idlestrategy"
+	"github.com/lirm/aeron-go/cluster/codecs"
+)
 
 type Cluster interface {
+	/**
+	 * Position the log has reached in bytes as of the current message.
+	 *
+	 * @return position the log has reached in bytes as of the current message.
+	 */
```
[the remark this answers is not in the document]
Comment (Aaronong) [reply cluster/cluster.go:14]: Makes perfect sense! Let me do that :)

[the remark this answers is not in the document]
Comment (Aaronong) [reply cluster/cluster.go]: @ethanf addressed comments! Thanks for the quick turnaround on the review

File: cluster/cluster.go [hunk 2]
```diff
@@ -1,15 +1,84 @@
...
-import "github.com/lirm/aeron-go/aeron/idlestrategy"
+import (
+	"github.com/lirm/aeron-go/aeron/atomic"
+	"github.com/lirm/aeron-go/aeron/idlestrategy"
+	"github.com/lirm/aeron-go/cluster/codecs"
+)
 
 type Cluster interface {
+
+	// Position returns the position the log has reached in bytes as of the current message.
```
Comment (ethanf) [inline cluster/cluster.go:11]: If you follow the style guide, the comment should start with the function name always

```
LogPosition returns the position the log has reached in bytes as of the current message.
```

**Review by ethanf — APPROVED** [review]

File: cluster/cluster.go (hunk 2 above)
Comment (Aaronong) [reply cluster/cluster.go]: Thanks for the feedback once again! Addressed!

File: cluster/cluster.go [hunk 3]
```diff
@@ -1,15 +1,84 @@
...
+	//
+	// Timers should only be scheduled or cancelled in the context of processing a
+	// ClusteredService#onSessionMessage(ClientSession, long, DirectBuffer, int, int, Header)
+	// ClusteredService#onTimerEvent(long, long)
+	// ClusteredService#onSessionOpen(ClientSession, long) or
+	// ClusteredService#onSessionClose(ClientSession, long, CloseReason)
+	// If applied to other events then they are not guaranteed to be reliable.
+	//
+	// Callers of this method should loop until the method succeeds, see {@link
+	// io.aeron.cluster.service.Cluster#scheduleTimer(long, long)} for an example.
```
Comment (ethanf) [inline cluster/cluster.go:64]: Lots of Java here too

Comment (Aaronong) [reply cluster/cluster.go]: apologies for the tardiness. Should be good now

**Comment by ethanf** [comment]
Can we clean up the comments to match the rest of the code? Other code doesn't do links to other functions, they don't comment on each parameter, etc.

**Comment by Aaronong** [comment]
Temporarily pulling PR for fixes

**Review by ethanf — APPROVED** [review]
Thanks for cleaning this up @Aaronong

**Pull request merged**